src/app/core/browse/browse-link-data.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@angular/core';
+import { BrowseDefinition } from '../shared/browse-definition.model';
+import { RequestService } from '../data/request.service';
+import { RemoteDataBuildService } from '../cache/builders/remote-data-build.service';
+import { ObjectCacheService } from '../cache/object-cache.service';
+import { HALEndpointService } from '../shared/hal-endpoint.service';
+import { Observable } from 'rxjs';
+import { RemoteData } from '../data/remote-data';
+import { PaginatedList } from '../data/paginated-list.model';
+import { IdentifiableDataService } from '../data/base/identifiable-data.service';
+import { FindAllDataImpl } from '../data/base/find-all-data';
+import { BrowseDefinitionDataService } from './browse-definition-data.service';
+import {
+  getFirstSucceededRemoteData, getPaginatedListPayload, getRemoteDataPayload
+} from '../shared/operators';
+import { distinctUntilChanged, map, startWith } from 'rxjs/operators';
+import { isEmpty, isNotEmpty } from '../../shared/empty.util';
+import { BrowseService } from './browse.service';
+
+/**
+ * Data service responsible for retrieving browse definitions from the REST server, IF AND ONLY IF
+ * they are configured as browse links (webui.browse.link.<n>)
+ */
+@Injectable()
+export class BrowseLinkDataService extends IdentifiableDataService<BrowseDefinition> {
```

Comment:
Hmm, looks like a review comment I made for the first version of this PR (but never submitted) was saved. This is no longer a problem since it was already addressed